### PR TITLE
Split CI into test/release

### PR DESCRIPTION
**Pull request by lepapareil**
push request and pull request now only launch tests jobs, tags now launches tests jobs and release jobs.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -69,58 +40,60 @@ jobs:
...
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: stable
+        toolchain: ${{ matrix.rust }}
         override: true
         profile: minimal
-    - name: Check Rust Version
+    - name: Build
       run: |
-        ci/check_version.sh
```
Comment (fabricereix) [inline .github/workflows/test.yml:85]: the check_version needs to be run each time, thus it is more suited for the test than release

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+name: releasing
```
Comment (fabricereix) [inline .github/workflows/release.yml:1]: name release everywhere

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,42 +1,13 @@
-name: CI
+name: testing
 
 on: [push, pull_request]
 
 env:
   CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+  VCPKGRS_DYNAMIC: 1
```
Comment (fabricereix) [inline .github/workflows/test.yml:8]: to be added with windows

**Comment by lepapareil** [comment]
hi @fabricereix, check version is not compatible with nigtly matrix. What do you suggest, separated jobs ?

**Comment by fabricereix** [comment]
as discussed, the check_version step can be removed (also delete ci/check_version.sh)
and additional debug info can be added at the start of the job including the current rust version being used.

**Comment by lepapareil** [comment]
check_version has been replaced by environment step on ci :) @fabricereix 

image

image

image

image

**Pull request merged**